examples/todo_example/server/plugin.ts

```diff
@@ -0,0 +1,368 @@
+        },
+      },
+      {
+        type: 'data_backfill',
+        backfillFn: (doc: SavedObjectModelTransformationDoc<Todo>) => {
+          if (doc.attributes.priority === undefined) {
+            doc.attributes.priority = 'Medium';
+          }
+          if (!doc.references) {
+            doc.references = [];
+          }
+          return doc;
+        },
+      },
+    ],
+  },
+};
+
+export class ToDoPlugin implements Plugin {
+  private readonly hideCompleted: boolean;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    const { hideCompleted } = initializerContext.config.get<{ hideCompleted: boolean }>();
+
+    this.hideCompleted = hideCompleted;
+  }
+
+  public setup(core: CoreSetup) {
+    const router = core.http.createRouter();
+    const savedObjects = core.savedObjects;
+
+    core.uiSettings.register({
+      'todo:hideCompleted': {
+        category: ['todo'],
+        description: i18n.translate('todo.uiSettings.hideCompleted.description', {
+          defaultMessage: 'Hide completed todos from the list.',
+        }),
+        name: i18n.translate('todo.uiSettings.hideCompleted.name', {
+          defaultMessage: 'Hide completed todos',
+        }),
+        requiresPageReload: false,
+        schema: schema.boolean(),
+        value: this.hideCompleted,
+      },
+    });
+
+    const SAVED_OBJECTS_TYPE = 'todo';
+    savedObjects.registerType({
+      mappings: {
+        properties: {
+          completed: { type: 'boolean' },
+          order: { type: 'integer' },
+          priority: { type: 'keyword' },
+          title: { type: 'text' },
+        },
+      },
+      modelVersions: MODEL_VERSIONS,
+      name: SAVED_OBJECTS_TYPE,
+      namespaceType: 'single',
+      hidden: false,
+    });
+
+    router.get(
+      {
+        path: '/api/todos',
+        options: { access: 'public' },
+        security: {
+          authz: {
+            enabled: false,
+            reason:
+              'This route is opted out of authorization because it is only intended for test use',
+          },
+        },
+        validate: false,
+      },
+      async (context, _request, response) => {
+        try {
+          const savedObjectsClient = (await context.core).savedObjects.getClient();
+          const uiSettingsClient = (await context.core).uiSettings.client;
+
+          const hideCompleted = await uiSettingsClient.get('todo:hideCompleted');
+
+          const todosData = await savedObjectsClient.find<Omit<Todo, 'id'>>({
+            type: SAVED_OBJECTS_TYPE,
+            sortField: 'order',
+            sortOrder: 'asc',
+          });
+
+          let todos = todosData.saved_objects.map(savedObjectToTodo);
+
+          if (hideCompleted) {
+            todos = todos.filter((todo) => !todo.completed);
+          }
+
+          return response.ok({ body: todos });
+        } catch (error) {
+          return response.badRequest({ body: error.message });
+        }
+      }
+    );
+
+    router.get(
```

Comment:
This is just a demo plugin so it's fine, just wanted to point out that we generally have routes in a separate directory (`server/routes` or the like)